addons/im_livechat/static/tests/channel_invite_tests.js

```diff
@@ -32,3 +32,70 @@ QUnit.test("Can invite a partner to a livechat channel", async (assert) => {
     await click("button[title='Show Member List']");
     assert.containsOnce($, ".o-discuss-ChannelMember:contains(James)");
 });
+
+QUnit.test("Available operators come first", async (assert) => {
+    const pyEnv = await startServer();
+    pyEnv["res.partner"].create({
+        name: "Harry",
+        im_status: "offline",
+        user_ids: [pyEnv["res.users"].create({ name: "Harry" })],
+    });
+    const availableOperatorId = pyEnv["res.partner"].create({
+        name: "Available operator",
+        im_status: "online",
+        user_ids: [pyEnv["res.users"].create({ name: "Available operator" })],
+    });
+    pyEnv["im_livechat.channel"].create({
+        available_operator_ids: [Command.create({ partner_id: availableOperatorId })],
+    });
+    const channelId = pyEnv["discuss.channel"].create({
+        anonymous_name: "Visitor #1",
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId }),
+            Command.create({ partner_id: pyEnv.publicPartnerId }),
+        ],
+        channel_type: "livechat",
+    });
+
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[title='Add Users']");
+    const partnerSuggestions = document.querySelectorAll(".o-discuss-ChannelInvitation-selectable");
+    assert.ok(partnerSuggestions[0].textContent.includes("Available operator"));
+    assert.ok(partnerSuggestions[1].textContent.includes("Harry"));
+});
+
+QUnit.test("Order on invitation count when none are available", async (assert) => {
+    const pyEnv = await startServer();
+    pyEnv["res.partner"].create({
+        name: "Less invited partner",
+        im_status: "offline",
+        user_ids: [pyEnv["res.users"].create({ name: "Less invited partner" })],
+    });
+    const mostInvitedPartnerId = pyEnv["res.partner"].create({
+        name: "Most invited partner",
+        im_status: "offline",
+        user_ids: [pyEnv["res.users"].create({ name: "Most invited partner" })],
+    });
+    pyEnv["discuss.channel"].create({
+        anonymous_name: "Visitor #1",
+        channel_member_ids: [
+            Command.create({ partner_id: mostInvitedPartnerId, create_uid: pyEnv.currentUserId }),
+        ],
+    });
+    const channelId = pyEnv["discuss.channel"].create({
+        anonymous_name: "Visitor #1",
+        channel_member_ids: [
+            Command.create({ partner_id: pyEnv.currentPartnerId }),
+            Command.create({ partner_id: pyEnv.publicPartnerId }),
+        ],
+        channel_type: "livechat",
+    });
+
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[title='Add Users']");
+    const partnerSuggestions = document.querySelectorAll(".o-discuss-ChannelInvitation-selectable");
+    assert.ok(partnerSuggestions[0].textContent.includes("Most invited partner"));
+    assert.ok(partnerSuggestions[1].textContent.includes("Less invited partner"));
+});
```

Comment:
I have no clue what this test really means functionally.
Test is poorly named (I don't know what it functionally means. "Order" of what? What's "invitation count"? What does "when none are available" means??
Name of partners "Most/Less invited" makes it even harder to read, and I don't see any clear reason for them to be named that way.

To make sure the test is written in a good way, the name of test should make it clear what are the functional intentions. The name of partners should be "John"/"Mitchell"/etc., and it should be easily understandable with just a glance at reading it.